File: oqyga_frontend/lib/features/user/domain/entities/profile_update_request.dart
import 'package:equatable/equatable.dart';

class ProfileUpdateRequest extends Equatable {
  final String? userName;
  final String? phoneNumber;
  final String? email;
  final String? userPhoto;

  const ProfileUpdateRequest({
    this.userName,
    this.phoneNumber,
    this.email,
    this.userPhoto,
  });

  @override
  List<Object?> get props => [userName, phoneNumber, email, userPhoto];
}

File: OqygaBackend/src/main/java/com/oqyga/OqygaBackend/dto/ProfileUpdateRequest.java
package com.oqyga.OqygaBackend.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Pattern;

@Data
public class ProfileUpdateRequest {

    @NotBlank(message = "Имя пользователя не может быть пустым")
    @Size(min = 3, max = 50, message = "Имя пользователя должно быть от 3 до 50 символов")
    private String userName;

    @Pattern(
            regexp = "^\\+7[0-9]{10}$",
            message = "Неверный формат номера телефона. Ожидается: +7(XXX) XXX XXXX"
    )
    private String phoneNumber;

    @Email(message = "Неверный формат почты")
    private String email;

    private String userPhoto;
}